src/js/components/molecules/faux-vertical-scroll/faux-vertical-scroll.ts

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable */
+import { Vue, Component, Prop } from 'vue-property-decorator';
+
+import { TweenLite, Power3 } from 'gsap';
+import Draggable from 'gsap/Draggable';
+import 'utils/throw-props-plugin';
+
+import Defaults from 'constants/defaults';
+import ViewportUtil from 'utils/viewport';
+import BrowserCapabilities from 'utils/browser-capabilities';
+
+@Component({
+    name: 'faux-vertical-scroll',
+    template: require('./faux-vertical-scroll.html'),
+})
+
+export default class FauxVerticalScroll extends Vue {
+
+    viewportUtil: ViewportUtil = new ViewportUtil();
+    transitionDuration: number = Defaults.defaultDuration;
+
+    draggableElement: any =  undefined;
+
+    scale: number = 0;
+    canDrag: boolean = false;
+    isMounted: boolean = false;
+
+    barStyle() {
+        return {
+            display: this.canDrag ? 'block' : 'hidden',
+            opacity: this.canDrag ? 1 : 0,
+            width: `${ this.scale }px`,
+        };
+    }
+
+    calculateOffset() {
```

Comment:
Can you add typescript on this function?

Also for the others functions like:
"tweenBar",
"createDraggable",
"calculateBarScale"